screens/UserScreens/UserProfile.jsx

```diff
@@ -1,16 +1,36 @@
-import { Button, Text, View } from "react-native";
-import { useState, useEffect } from "react";
+import { View, TouchableOpacity, Text, ScrollView, Modal } from "react-native";
 import { signOut } from "firebase/auth";
-import { get, ref, child } from "firebase/database";
-import { firebase_auth, firebase_db } from "../../firebase.config.js";
+import { useState, useEffect } from "react";
+import { updateProfile } from "firebase/auth";
+import { ref, uploadBytes, getDownloadURL } from "firebase/storage";
+import { get, child, ref as dbRef } from "firebase/database";
+import { launchImageLibraryAsync } from "expo-image-picker";
+import { firebase_auth, firebase_db, storage } from "../../firebase.config.js";
+import ProfilePanel from "../../components/ProfileElements/ProfilePanel.jsx";
+import UserDetailsModal from "../../components/ProfileElements/UserDetailsModal.jsx";
+import {
+    faCircleExclamation,
+    faCircleInfo,
+    faPencil,
+    faPhone,
+    faRightFromBracket,
+    faTrash,
+} from "@fortawesome/free-solid-svg-icons";
+import ProfilePicture from "../../components/ProfileElements/ProfilePicture.jsx";
+import ContactInformationModal from "../../components/ProfileElements/ContactInformationModal.jsx";
+import ReportErrorModal from "../../components/ProfileElements/ReportErrorModal.jsx";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+
 
 export default function UserProfileScreen({ navigation }) {
     const [userName, setUserName] = useState("");
-    const auth = firebase_auth;
-    const db = firebase_db;
-    
+    let [modal, setModal] = useState(null);
+    let auth = firebase_auth;
+    let db = firebase_db;
+    const storageRef = ref(storage);
+
     useEffect(() => {
-        get(child(ref(db), `users/${auth.currentUser.uid}`)).then((snapshot) => {
+        get(child(dbRef(db), `users/${auth.currentUser.uid}`)).then((snapshot) => {
```

Comment:
bit confused by mix of ref (4 lines up) and dbRef?

Reply:
Spoken about